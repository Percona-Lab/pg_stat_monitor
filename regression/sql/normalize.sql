CREATE EXTENSION pg_stat_monitor;
SET pg_stat_monitor.pgsm_normalized_query TO off;

CREATE TABLE tb1(id int, name text);
INSERT INTO tb1 VALUES (1,'foo'),(2,'bar'),(3,'baz');

CREATE OR REPLACE FUNCTION dup(int) RETURNS TABLE(f1 int, f2 text)
AS $$ SELECT $1, CAST($1 AS text) || ' is text' $$
LANGUAGE SQL;

PREPARE gettb1 AS SELECT name,'test placeholder in string: $1' AS test_string FROM tb1 WHERE id = $1;
SELECT pg_stat_monitor_reset();

SELECT dup(22);
EXECUTE gettb1(2);

SELECT query, calls FROM pg_stat_monitor ORDER BY query COLLATE "C";
DROP TABLE tb1;
DROP FUNCTION dup;
DEALLOCATE gettb1;
SELECT pg_stat_monitor_reset();
DROP EXTENSION pg_stat_monitor;