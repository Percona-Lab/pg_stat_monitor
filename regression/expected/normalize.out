CREATE EXTENSION pg_stat_monitor;
SET pg_stat_monitor.pgsm_normalized_query TO off;
CREATE TABLE tb1(id int, name text);
INSERT INTO tb1 VALUES (1,'foo'),(2,'bar'),(3,'baz');
CREATE OR REPLACE FUNCTION dup(int) RETURNS TABLE(f1 int, f2 text)
AS $$ SELECT $1, CAST($1 AS text) || ' is text' $$
LANGUAGE SQL;
PREPARE gettb1 AS SELECT name,'test placeholder in string: $1' AS test_string FROM tb1 WHERE id = $1;
SELECT pg_stat_monitor_reset();
 pg_stat_monitor_reset 
-----------------------
 
(1 row)

SELECT dup(22);
        dup        
-------------------
 (22,"22 is text")
(1 row)

EXECUTE gettb1(2);
 name |          test_string           
------+--------------------------------
 bar  | test placeholder in string: $1
(1 row)

SELECT query, calls FROM pg_stat_monitor ORDER BY query COLLATE "C";
                                                query                                                 | calls 
------------------------------------------------------------------------------------------------------+-------
  SELECT 22, CAST(22 AS text) || ' is text'                                                           |     1
 <insufficient shared space or query text not set yet>                                                |     1
 PREPARE gettb1 AS SELECT name,'test placeholder in string: $1' AS test_string FROM tb1 WHERE id = 2; |     1
 SELECT dup(22);                                                                                      |     1
 SELECT pg_stat_monitor_reset();                                                                      |     1
(5 rows)

DROP TABLE tb1;
DROP FUNCTION dup;
DEALLOCATE gettb1;
SELECT pg_stat_monitor_reset();
 pg_stat_monitor_reset 
-----------------------
 
(1 row)

DROP EXTENSION pg_stat_monitor;
